core/src/main/scala/org/apache/spark/ExecutorAllocationManager.scala

```diff
@@ -262,15 +267,22 @@ private[spark] class ExecutorAllocationManager(
     val maxNeeded = maxNumExecutorsNeeded
 
     if (maxNeeded < numExecutorsTarget) {
-      // The target number exceeds the number we actually need, so stop adding new
-      // executors and inform the cluster manager to cancel the extra pending requests
-      val oldNumExecutorsTarget = numExecutorsTarget
-      numExecutorsTarget = math.max(maxNeeded, minNumExecutors)
-      client.requestTotalExecutors(numExecutorsTarget)
-      numExecutorsToAdd = 1
-      logInfo(s"Lowering target number of executors to $numExecutorsTarget because " +
-        s"not all requests are actually needed (previously $oldNumExecutorsTarget)")
-      numExecutorsTarget - oldNumExecutorsTarget
+      if (!numTargetExecutorAdjustable.get) {
+        // Keep the initial number of target executor to not ramp down until the first job is
+        // submitted or the first idle executor is released.
+        client.requestTotalExecutors(numExecutorsTarget)
```

Comment:
Actually, does it make sense to do this inside the case `maxNeeded < numExecutorsTarget`? This condition implies that we are targeting more executors that are needed, and so we should cancel outstanding requests. This is not really true during initialization.

I think a better indication that your application has initialized is if the number of executors is >= your initial executors.